stream-chat-android-compose/src/main/java/io/getstream/chat/android/compose/ui/imagepreview/ImagePreviewActivity.kt

```diff
@@ -141,35 +265,245 @@ public class ImagePreviewActivity : AppCompatActivity() {
                 color = ChatTheme.colors.textHighEmphasis
             )
 
+            Timestamp(date = message.updatedAt ?: message.createdAt ?: Date())
+        }
+    }
+
+    /**
+     * Toggles the image options menu.
+     *
+     * @param modifier Modifier for styling.
+     */
+    @Composable
+    private fun ImagePreviewOptionsToggle(
+        modifier: Modifier = Modifier,
+    ) {
+        Icon(
+            modifier = modifier
+                .size(24.dp)
+                .clickable(
+                    interactionSource = remember { MutableInteractionSource() },
+                    indication = rememberRipple(bounded = false),
+                    onClick = { imagePreviewViewModel.toggleImageOptions(isShowingOptions = true) }
+                ),
+            imageVector = Icons.Default.MoreVert,
+            contentDescription = stringResource(R.string.stream_compose_image_options),
+            tint = ChatTheme.colors.textHighEmphasis
+        )
+    }
+
+    /**
+     * The image options menu, used to perform different actions for the currently active image.
+     *
+     * @param options The options available for the image.
+     * @param pagerState The state of the pager, used to fetch the current image.
+     * @param modifier Modifier for styling.
+     */
+    @Composable
+    private fun ImagePreviewOptions(
+        options: List<ImagePreviewOption>,
+        pagerState: PagerState,
+        modifier: Modifier,
+    ) {
+        Box(
+            modifier = Modifier
+                .fillMaxSize()
+                .background(ChatTheme.colors.overlay)
+                .clickable(
+                    indication = null,
+                    interactionSource = remember { MutableInteractionSource() },
+                    onClick = { imagePreviewViewModel.toggleImageOptions(isShowingOptions = false) }
+                )
+        ) {
+            Surface(
+                modifier = modifier
+                    .padding(16.dp)
+                    .width(150.dp)
+                    .wrapContentHeight()
+                    .align(Alignment.TopEnd),
+                shape = RoundedCornerShape(16.dp),
+                elevation = 4.dp,
+                color = ChatTheme.colors.barsBackground
+            ) {
+                Column(modifier = Modifier.fillMaxWidth()) {
+                    options.forEachIndexed { index, option ->
```

Comment:
Not worth using a `LazyColumn` for? Wouldn't we get a nicer solution for the spacers between items if we used that?

Reply:
I think that it would be a bit of an overkill here, as we're just looking for a (generally) static list of a few items.

In terms of the spacers - these are dividers, so we'd have to do something similar here anyway.

So I figured it might've been confusing to have a LazyColumn and have all the items loaded eagerly.